src/go/k8s/pkg/resources/statefulset_update.go

```diff
@@ -420,6 +457,10 @@ func (e *RequeueAfterError) Error() string {
 	return fmt.Sprintf("RequeueAfterError %s", e.Msg)
 }
 
+func (e *RequeueAfterError) Is(target error) bool {
+	return e.Error() == target.Error()
```

Comment:
do you really want to use `==` and not rather `errors.Is`?

Reply:
Hmm. I implemented the `Is` function as I was not able to unit test that error. I can check on the side if that would work, but it should work in the first place in direct call.